From: Keith Busch <keith.busch@intel.com>
Date: Thu, 18 Feb 2016 14:56:35 -0700
Subject: blk-mq: Fix NULL pointer updating nr_requests
Git-commit: e9137d4b93078b6a9965acfb18a2a2ad91cf8405
Patch-mainline: v4.6-rc1
References: FATE#319965, bsc#964944

A h/w context's tags are freed if it was not assigned a CPU. Check if
the context has tags before updating the depth.

Signed-off-by: Keith Busch <keith.busch@intel.com>
Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.com>
---
 block/blk-mq.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -2337,6 +2337,8 @@ int blk_mq_update_nr_requests(struct req
 
 	ret = 0;
 	queue_for_each_hw_ctx(q, hctx, i) {
+		if (!hctx->tags)
+			continue;
 		ret = blk_mq_tag_update_depth(hctx->tags, nr);
 		if (ret)
 			break;
